### PR TITLE
Add status output for Nitrokey Pro

### PR DESCRIPTION
**Pull request by robinkrahl**

### CODE REVIEW

File: nitrocli/src/commands.rs [hunk 1]
```diff
@@ -171,41 +174,30 @@ where
...
+/// Query and pretty print the common status of all Nitrokey devices.
+fn print_status(model: &'static str, device: &nitrokey::DeviceWrapper) -> Result<()> {
+  let serial_number = device
+    .get_serial_number()
+    .map_err(|err| get_error("Could not query the serial number", &err))?;
   println!(
     r#"Status:
   model:             {model}
-  smart card ID:     {id}
+  smart card ID:     0x{id}
```
Comment (d-e-s-o) [inline nitrocli/src/commands.rs]: Nit: Can this addition be moved into the previous commit where you factored out this function? Or am I missing something?

File: nitrocli/src/commands.rs [hunk 2]
```diff
@@ -171,41 +174,30 @@ where
...
   firmware version:  {fwv0}.{fwv1}
   user retry count:  {urc}
   admin retry count: {arc}"#,
     model = model,
-    id = smartcard_id,
-    fwv0 = firmware_major,
-    fwv1 = firmware_minor,
-    urc = user_retries,
-    arc = admin_retries,
+    id = serial_number,
```
Comment (d-e-s-o) [inline nitrocli/src/commands.rs]: Does it make sense to differentiate between the smart card ID (for the storage device) and the serial number for the Pro? The string `smart card ID` seems to be misleading now, and I am trying to understand whether to move this part into the device specific print function or keep it here and change the string to `serial number`.

File: nitrocli/src/commands.rs (hunk 1 above)
Comment (robinkrahl) [reply nitrocli/src/commands.rs]: No.  This change is needed as we replace the value from the `StorageStatus` struct (with leading `0x`) with the value from `Device::get_serial_number` (no leading `0x`) in this commit.

File: nitrocli/src/commands.rs (hunk 2 above)
Comment (robinkrahl) [reply nitrocli/src/commands.rs]: I’d prefer to have this as a common field.  It is the same ID that we will show in the pinentry dialog (see #24) and that is reported by `gpg --card-status` in the same way for both models.   I’ll rename it to *serial number*.

That said, for my Nitrokey Storage, the serial number returned by `NK_get_status_storage` and `NK_device_serial_number` do not match the number reported by `gpg --card-status`.  I’ll have to investigate that.  Could you please check the result of `NK_device_serial_number` for your Storage stick?

**Comment by robinkrahl** [comment]
> I believe you should also adjust the help message printed for the status command, which still talks about the storage command only.

Thanks!  I  missed that.

File: nitrocli/src/commands.rs (hunk 2 above)
Comment (d-e-s-o) [reply nitrocli/src/commands.rs]: Sorry, I did not get to that today. I will get back to you tomorrow.

Comment (robinkrahl) [reply nitrocli/src/commands.rs]: I opened Nitrokey/nitrokey-storage-firmware#76 for this issue.

Comment (robinkrahl) [reply nitrocli/src/commands.rs]: No, I’ve never experienced such behavior.  But I’ve never used GnuPG on my development stick, and I only occasionally run `nitrocli` with my production stick.  And I do not use the Storage volumes at all.  So there might be interaction problems that I did not trigger with my usage pattern.  (That reminds me that we have to implement a `--debug` option that enables the debugging output from `libnitrokey`.)

**Comment by d-e-s-o** [comment]
Merged. Thanks Robin!

**Pull request closed without merging**